### PR TITLE
Clarify how to find changelog for the RC candidates of providers

### PR DESCRIPTION
**Pull request by potiuk**
We used to have changelog in wheel package description but since those changelogs are immutable in the released packages, we removed the changelog from description some time ago and instead we publish link to the changelog as "project URL". However the changelog links are not available until we publish the packages, so there is no easy way to see the changelog - and the "Status of providers" is the easiest (and pretty convenient) way of looking at what changes are coming.

This PR removes mentioning that "changelogs are available" in the PyPI packages and explains that you should look for included PRs in the "Status issue" until the providers are released.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by hussein-awala — APPROVED** [review]
👍

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 1]
```diff
@@ -649,7 +648,11 @@ the artifact checksums when we actually release.
 The status of testing the providers by the community is kept here:
 <TODO COPY LINK TO THE ISSUE CREATED>
 
-You can find packages as well as detailed changelog following the below links:
+The issue is also the easiest way to see which code affecting PRs are included in the RC candidates.
```
Comment (hussein-awala) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md]: Nit: It would be better to rephrase `which code affecting PRs are included`

**Review by dirrao — APPROVED** [review]

File: dev/README_RELEASE_PROVIDER_PACKAGES.md (hunk 1 above)
Comment (potiuk) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md]:
```suggestion
The issue is also the easiest way to see important PRs included in the RC candidates.
```

Comment (potiuk) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md]: Rephrased.

**Review by jscheffl — APPROVED** [review]

**Pull request merged**